diff --git a/v2rayN/v2rayN/Views/ProfilesView.xaml.cs b/v2rayN/v2rayN/Views/ProfilesView.xaml.cs
index 47683b317d2..9022c99f55b 100644
--- a/v2rayN/v2rayN/Views/ProfilesView.xaml.cs
+++ b/v2rayN/v2rayN/Views/ProfilesView.xaml.cs
@@ -346,17 +346,18 @@ private void RestoreUI()
                             item2.Width = item.Width;
                             item2.DisplayIndex = displayIndex++;
                         }
-                        if (item.Name.StartsWith("to", StringComparison.CurrentCultureIgnoreCase))
-                        {
-                            item2.Visibility = _config.GuiItem.EnableStatistics ? Visibility.Visible : Visibility.Hidden;
-                        }
-                        if (item.Name.Equals("IpInfo", StringComparison.CurrentCultureIgnoreCase))
-                        {
-                            item2.Visibility = _config.SpeedTestItem.IPAPIUrl.IsNotEmpty() && !_config.UiItem.HideColumnIpInfo ? Visibility.Visible : Visibility.Hidden;
-                        }
                     }
                 }
             }
+
+            foreach (var c in lstProfiles.Columns.Cast<MyDGTextColumn>())
+            {
+                if (c.ExName.StartsWith("to", StringComparison.CurrentCultureIgnoreCase))
+                    c.Visibility = _config.GuiItem.EnableStatistics ? Visibility.Visible : Visibility.Hidden;
+                else if (c.ExName.Equals("IpInfo", StringComparison.CurrentCultureIgnoreCase))
+                    c.Visibility = _config.SpeedTestItem.IPAPIUrl.IsNotEmpty() && !_config.UiItem.HideColumnIpInfo ? Visibility.Visible : Visibility.Hidden;
+            }
+
         }
         catch (Exception ex)
         {
