diff --git a/package-osx.sh b/package-osx.sh
index 91a0fa3c9..5f51c261d 100755
--- a/package-osx.sh
+++ b/package-osx.sh
@@ -7,6 +7,10 @@ Version="$3"
 FileName="v2rayN-${Arch}.zip"
 wget -nv -O $FileName "https://github.com/2dust/v2rayN-core-bin/raw/refs/heads/master/$FileName"
 7z x $FileName
+
+rm -rf "v2rayN-${Arch}/bin/xray"
+rm -rf "v2rayN-${Arch}/bin/mihomo"
+
 cp -rf v2rayN-${Arch}/* $OutputPath
 
 PackagePath="v2rayN-Package-${Arch}"
