diff --git a/v2rayN/ServiceLib/Manager/CoreInfoManager.cs b/v2rayN/ServiceLib/Manager/CoreInfoManager.cs
index bda3108ae44..caa792fcafd 100644
--- a/v2rayN/ServiceLib/Manager/CoreInfoManager.cs
+++ b/v2rayN/ServiceLib/Manager/CoreInfoManager.cs
@@ -54,7 +54,7 @@ public List<ECoreType> GetCheckUpdateCoreTypes()
     {
         var lst = new List<ECoreType>();
 
-        if (RuntimeInformation.ProcessArchitecture != Architecture.X86)
+        if (RuntimeInformation.ProcessArchitecture != Architecture.Arm)
         {
             if (IsCheckUpdateSupported(ECoreType.v2rayN))
             {
@@ -63,8 +63,8 @@ public List<ECoreType> GetCheckUpdateCoreTypes()
 
             if (!(Utils.IsWindows() && Environment.OSVersion.Version.Major < 10))
             {
-                lst.Add(ECoreType.Xray);
-                lst.Add(ECoreType.mihomo);
+                // lst.Add(ECoreType.Xray);
+                // lst.Add(ECoreType.mihomo);
                 lst.Add(ECoreType.sing_box);
             }
         }
@@ -110,6 +110,8 @@ private void InitCoreInfo()
                     ReleaseApiUrl = urlN.Replace(Global.GithubUrl, Global.GithubApiUrl),
                     DownloadUrlWin64 = urlN + "/download/{0}/v2rayN-windows-64.zip",
                     DownloadUrlWinArm64 = urlN + "/download/{0}/v2rayN-windows-arm64.zip",
+                    DownloadUrlWin86 = urlN + "/download/{0}/v2rayN-windows-86.zip",
+                    DownloadUrlLinuxPpc64le = urlN + "/download/{0}/v2rayN-linux-ppc64le.zip",
                     DownloadUrlLinux64 = urlN + "/download/{0}/v2rayN-linux-64.zip",
                     DownloadUrlLinuxArm64 = urlN + "/download/{0}/v2rayN-linux-arm64.zip",
                     DownloadUrlLinuxRiscV64 = urlN + "/download/{0}/v2rayN-linux-riscv64.zip",
@@ -223,6 +225,8 @@ private void InitCoreInfo()
                     ReleaseApiUrl = urlSingbox.Replace(Global.GithubUrl, Global.GithubApiUrl),
                     DownloadUrlWin64 = urlSingbox + "/download/{0}/sing-box-{1}-windows-amd64.zip",
                     DownloadUrlWinArm64 = urlSingbox + "/download/{0}/sing-box-{1}-windows-arm64.zip",
+                    DownloadUrlWin86 = urlSingbox + "/download/{0}/sing-box-{1}-windows-386.zip",
+                    DownloadUrlLinuxPpc64le = urlSingbox + "/download/{0}/sing-box-{1}-linux-ppc64le.tar.gz",
                     DownloadUrlLinux64 = urlSingbox + "/download/{0}/sing-box-{1}-linux-amd64.tar.gz",
                     DownloadUrlLinuxArm64 = urlSingbox + "/download/{0}/sing-box-{1}-linux-arm64.tar.gz",
                     DownloadUrlLinuxRiscV64 = urlSingbox + "/download/{0}/sing-box-{1}-linux-riscv64.tar.gz",
