diff --git a/v2rayN/ServiceLib/Manager/CoreInfoManager.cs b/v2rayN/ServiceLib/Manager/CoreInfoManager.cs
index de5875e58bc..541c9e948da 100644
--- a/v2rayN/ServiceLib/Manager/CoreInfoManager.cs
+++ b/v2rayN/ServiceLib/Manager/CoreInfoManager.cs
@@ -90,6 +90,7 @@ public bool GetCheckPreRelease(ECoreType type, bool preRelease)
         {
             ECoreType.v2rayN => preRelease,
             ECoreType.Xray => preRelease,
+            ECoreType.sing_box => preRelease,
             _ => false,
         };
     }
